ts/Core/Dynamics.ts

```diff
@@ -1667,6 +1671,33 @@ extend(Series.prototype, /** @lends Series.prototype */ {
     setName: function (this: Highcharts.Series, name: string): void {
         this.name = this.options.name = this.userOptions.name = name;
         this.chart.isDirtyLegend = true;
+    },
+    /**
+     * Check if the option has changed.
+     *
+     * @private
+     * @function Highcharts.Series#hasOptionChanged
+     *
+     * @param {string} option
+     *
+     * @return {boolean}
+     */
+    hasOptionChanged(this: Highcharts.Series, optionName: string): boolean {
```

Comment:
This method works nicely for primitive properties (string, number etc), but won't work with arrays/objects. Comparing `{ x: 5 }` with `{ x: 5 }` won't work (`!==` compares references), unless we implement deep recursive comparison. That seems to be an overhead for `dataGrouping`/`keys` option only. Regarding `keys` see my disclaimer below, but `dataGrouping` is related to Stock only, not Highcharts. Perhaps Stock should overwrite this methods (or use events)?

Anyway, I see a few possible solutions here:
- ignore that updating `plotOptions.series.dataGrouping` doesn't work -> as already in current release, no changes then
- merge this branch, but this affects performance -> `{} !== {}` so points are recreated always when `dataGrouping` is set like this `series: [{ dataGrouping: { ... } }]`. Sole purpose of `dataGrouping` is to increase the performance so this solution should be avoided
- implement deep copy (in Highcharts or in Stock) -> resolves the issues, but produces extra code. And we end up with another recursive function to maintain (`merge()`, `cleanRecursive()`, `extend()` and something in tooltip and in networkgraph and who knows where else :))
- implement simple comparison of objects/arrays - limited to the first level only (instead of recursive solution) -> could be lightweight solution, with just a few lines of code

Disclaimer: I said "it won't work with arrays" but it's not true. It works, because our `merge()` does copy arrays (and nodes afaik) directly, not item by item (performance reasons) :)

What do you think @oysteinmoseng @sebastianbochan @bre1470 ? Any other ideas? Which one do you prefer?

Reply:
Ping @oysteinmoseng @sebastianbochan @bre1470 and now also @TorsteinHonsi :)

Reply:
I would suggest writing some specific performance tests so we're not shooting in the dark here. It can be hard to predict what will be performance bottlenecks. If I understand the issue correctly the ideal solution would be that `hasOptionChanged` does a deep comparison between objects (and arrays?). This might not necessarily be a noticeable performance hit at all. We don't see deep nested objects and arrays passed to `series.update` normally.

Not a big fan of only solving this in Stock since we are probably eventually going to add things to plotOptions in Highcharts that requires deep comparisons too (if we don't have some already?).

Leaving it as is (maybe documenting it) is the pragmatic approach of course, until someone complains.

Reply:
I would say that `implement simple comparison of objects/arrays` sounds like a good plan and an usefull tool also for other issues. 

Additionally, it can be useful in a bug like: exporting (compare userOptions and options for an axis).

Reply:
@TorsteinHonsi Do you have any thoughts, about choosing the right solution?

Reply:
Actually, after trying different approaches, I like this one best. 

> this affects performance -> {} !== {} so points are recreated always when dataGrouping is set like this series: [{ dataGrouping: { ... } }]

I think it is okay that points are always recreate in this scenario. When `dataGrouping` is set in the updated options, we must assume it is because it should be changed.